test/unit/services/shopify_api_test.rb

```diff
@@ -35,6 +35,19 @@ def test_request_uri_is_correct_when_callback_url_has_trailing_slash
     end
   end
 
+  def test_request_uri_is_correct_when_callback_url_has_additional_element
+    Timecop.freeze do
+      service = build_service(callback_url: 'https://dotcom-prod.bluemercuryio.com/v1/dotcom')
+      timestamp = Time.now.utc.to_i
+      uri = service.send(:request_uri, 'fetch_stock', {sku: '123', timestamp: timestamp, shop: 'www.snowwowdevil.ca'})
+      assert_equal "https://dotcom-prod.bluemercuryio.com/v1/dotcom/fetch_stock.json?shop=www.snowwowdevil.ca&sku=123&timestamp=#{timestamp}", uri.to_s
+
+      service = build_service(callback_url: 'http://supershopifyapptwin.com/shopify')
```

Comment:
I think this half of the test is redundant, to my understanding it doesn't catch anything that wouldn't have been caught by the first half (asserting that the correct callback URL is used even with extra components).